pjsip/src/pjsua-lib/pjsua_call.c

```diff
@@ -5576,13 +5576,18 @@ static void pjsua_call_on_tsx_state_changed(pjsip_inv_session *inv,
 		found_idx = pj_strtok(&input, &delim, &token, 0);
 		if (found_idx != input.slen) {
 		    /* Get signal/digit */
-		    const pj_str_t STR_SIGNAL = { "Signal=", 7 };
-		    const pj_str_t STR_DURATION = { "Duration=", 9 };
+		    const pj_str_t STR_SIGNAL = { "Signal", 6 };
+		    const pj_str_t STR_DURATION = { "Duration", 8 };
 		    char *val;
 
 		    val = pj_strstr(&input, &STR_SIGNAL);
 		    if (val) {
-			info.digit = *(val+STR_SIGNAL.slen);
+				pj_ssize_t idx1 = val - input.ptr;
+				pj_ssize_t idx2 = 0;
+				while ( (*(val + STR_SIGNAL.slen + idx2) == 32 || *(val + STR_SIGNAL.slen + idx2) == 61) &&
+					(idx1 + idx2 < input.slen))
```

Comment:
shouldn't loop condition `(idx1 + idx2 < input.slen)` include `STR_SIGNAL.slen`, i.e: `(idx1 + STR_SIGNAL.slen + idx2 < input.slen)`?
or alternatively, initialize `idx1` with `val - input.ptr + STR_SIGNAL.slen`?

Reply:
Yes, It is better.